icu4c/source/common/uniset_props.cpp

```diff
@@ -105,13 +105,35 @@ static UBool U_CALLCONV uset_cleanup(void) {
     return TRUE;
 }
 
+#ifdef __SUNPRO_CC
+static void U_CALLCONV createUni32Set(UErrorCode &errorCode) {
```

Comment:
It looks like you're creating duplicated code.  Maybe just move the code here on all platforms?